client/my-sites/plans-features-main/hooks/use-generate-action.ts

```diff
@@ -0,0 +1,298 @@
+			textOverride = translate( 'Get Performance', { textOnly: true } );
+		} else if (
+			isWooExpressSmallPlan( planSlug ) &&
+			! isWooExpressSmallPlan( sitePlanSlug || '' )
+		) {
+			textOverride = translate( 'Get Essential', { textOnly: true } );
+		} else if ( isBusinessTrial( sitePlanSlug || '' ) ) {
+			textOverride = translate( 'Get %(plan)s', {
+				textOnly: true,
+				args: {
+					plan: getPlan( planSlug )?.getTitle() || '',
+				},
+			} );
+		}
+
+		if ( isFreePlan( planSlug ) ) {
+			text = translate( 'Contact support', { context: 'verb' } );
+
+			// TODO: Consider DRYing this up
+			if ( sitePlanSlug === planSlug && intentFromProps !== 'plans-p2' ) {
+				text = translate( 'Manage add-ons', { context: 'verb' } );
+				status = 'enabled';
+			}
+		} else if (
+			availableForPurchase &&
+			sitePlanSlug &&
+			! current &&
+			! isTrialPlan &&
+			currentPlanBillingPeriod &&
+			billingPeriod &&
+			currentPlanBillingPeriod > billingPeriod
+		) {
+			text = translate( 'Contact support', { context: 'verb' } );
+		} else if (
+			availableForPurchase &&
+			sitePlanSlug &&
+			! current &&
+			getPlanClass( planSlug ) === getPlanClass( sitePlanSlug ) &&
+			! isTrialPlan
+		) {
+			// If the current plan matches on a lower-term, then show an "Upgrade to..." button.
+			if ( planMatches( planSlug, { term: TERM_TRIENNIALLY } ) ) {
+				text = textOverride || translate( 'Upgrade to Triennial' );
+			}
+
+			if ( planMatches( planSlug, { term: TERM_BIENNIALLY } ) ) {
+				text = textOverride || translate( 'Upgrade to Biennial' );
+			}
+
+			if ( planMatches( planSlug, { term: TERM_ANNUALLY } ) ) {
+				text = textOverride || translate( 'Upgrade to Yearly' );
+			}
+		} else if ( sitePlanSlug === planSlug && intentFromProps !== 'plans-p2' ) {
+			// All other actions for a current plan
+			text = translate( 'View plan' );
+
+			if ( canUserManageCurrentPlan ) {
+				text = translate( 'Manage plan' );
+			} else if ( domainFromHomeUpsellFlow ) {
+				text = translate( 'Keep my plan', { context: 'verb' } );
+			}
+		} else if ( isStuck && availableForPurchase ) {
+			// If the ctas are stickied, show the full price
+			text = translate( 'Upgrade – %(priceString)s', {
+				context: 'verb',
+				args: { priceString: priceString ?? '' },
+				comment: '%(priceString)s is the full price including the currency. Eg: Get Upgrade - $10',
+			} );
+
+			if ( isLargeCurrency ) {
+				text = translate( 'Get %(plan)s {{span}}%(priceString)s{{/span}}', {
+					args: {
+						plan: planTitle,
+						priceString: priceString ?? '',
+					},
+					comment:
+						'%(plan)s is the name of the plan and %(priceString)s is the full price including the currency. Eg: Get Premium - $10',
+					components: {
+						// span: <span className="plan-features-2023-grid__actions-signup-plan-text" />,
+					},
+				} );
+			}
+		} else if ( ! availableForPurchase ) {
+			text = translate( 'Downgrade', { context: 'verb' } );
```

Comment:
I wonder if it might be best to set `text` to `null` here (for now) and leave the entire handling done in `actions.tsx` (since it adds a tooltip and also a different text for mobile). We can decide later how to abstract that part. Right now it's the same logic (and texts) split in half.